Core/include/Acts/Utilities/detail/Subspace.hpp

```diff
@@ -204,6 +210,175 @@ class FixedSizeSubspace {
+  constexpr bool contains(std::size_t index) const {
+    bool isContained = false;
+    // always iterate over all elements to avoid branching and hope the compiler
+    // can optimise this for us.
+    for (std::size_t i = 0; i < kFullSize; ++i) {
+      isContained |= ((i < m_size) & (m_axes[i] == index));
+    }
+    return isContained;
+  }
+
+  /// Projection matrix that maps from the full space into the subspace.
+  ///
+  /// @tparam scalar_t Scalar type for the projection matrix
+  template <typename scalar_t>
+  auto projector() const -> ProjectionMatrix<scalar_t> {
+    ProjectionMatrix<scalar_t> proj(m_size, kFullSize);
+    proj.setZero();
+    for (auto i = 0u; i < m_size; ++i) {
+      proj(i, m_axes[i]) = 1;
+    }
+    return proj;
+  }
+
+  /// Expansion matrix that maps from the subspace into the full space.
+  ///
+  /// @tparam scalar_t Scalar type of the generated expansion matrix
+  template <typename scalar_t>
+  auto expander() const -> ExpansionMatrix<scalar_t> {
+    ExpansionMatrix<scalar_t> expn(kFullSize, m_size);
+    expn.setZero();
+    for (auto i = 0u; i < m_size; ++i) {
+      expn(m_axes[i], i) = 1;
+    }
+    return expn;
+  }
+
+  /// Projection matrix that maps from the full space into the subspace.
+  ///
+  /// @tparam scalar_t Scalar type for the projection matrix
+  template <typename scalar_t>
+  auto fullProjector() const -> FullProjectionMatrix<scalar_t> {
+    FullProjectionMatrix<scalar_t> proj;
+    proj.setZero();
+    for (auto i = 0u; i < m_size; ++i) {
+      proj(i, m_axes[i]) = 1;
+    }
+    return proj;
+  }
+
+  /// Expansion matrix that maps from the subspace into the full space.
+  ///
+  /// @tparam scalar_t Scalar type of the generated expansion matrix
+  template <typename scalar_t>
+  auto fullExpander() const -> FullExpansionMatrix<scalar_t> {
+    FullExpansionMatrix<scalar_t> expn;
+    expn.setZero();
+    for (auto i = 0u; i < m_size; ++i) {
+      expn(m_axes[i], i) = 1;
+    }
+    return expn;
+  }
+
+  std::uint64_t projectorBits() const {
+    std::uint64_t result = 0;
+
+    for (std::size_t i = 0; i < m_size; ++i) {
+      for (std::size_t j = 0; j < kFullSize; ++j) {
+        // the bit order is defined in `Acts/Utilities/AlgebraHelpers.hpp`
+        // in `matrixToBitset`
+        std::size_t index = m_size * kFullSize - 1 - (i + j * m_size);
+        if (m_axes[i] == j) {
+          result |= (1ull << index);
+        }
+      }
+    }
+
+    return result;
+  }
+
+  std::uint64_t fullProjectorBits() const {
+    std::uint64_t result = 0;
+
+    for (std::size_t i = 0; i < kFullSize; ++i) {
+      for (std::size_t j = 0; j < kFullSize; ++j) {
+        // the bit order is defined in `Acts/Utilities/AlgebraHelpers.hpp`
+        // in `matrixToBitset`
+        std::size_t index = kFullSize * kFullSize - 1 - (i + j * kFullSize);
+        if (i < m_size && m_axes[i] == j) {
+          result |= (1ull << index);
```

Comment:
Same here.

Reply:
Here it is an actual bitwise operation. Does it also complain about that?